ldap/data_source_resource_ldap_user.go

```diff
@@ -0,0 +1,80 @@
+package ldap
+
+import (
+	"github.com/Ouest-France/goldap"
+	"github.com/go-ldap/ldap/v3"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceLDAPUser() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceLDAPUserRead,
+		Schema: map[string]*schema.Schema{
+			"ou": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"sam_account_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"user_principal_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func dataSourceLDAPUserRead(d *schema.ResourceData, m interface{}) error {
+	return resourceLDAPUserRead(d, m)
```

Comment:
With the context-aware Read method we have to pass context, and we can use it to signal that the call is from a datasource like in group one: https://github.com/Ouest-France/terraform-provider-ldap/blob/master/ldap/data_source_resource_ldap_group.go#L45

Reply:
right. fixed.